### ISSUE TITLE
Extra spacing between Hangul and other scripts?

### ISSUE THREAD

**Issue by xfq** [w3c/klreq#58]
When discussing https://github.com/w3c/csswg-drafts/issues/9979 , we have two questions about the spacing between Hangul and other characters:

1. Should there be extra spacing between Hangul and Hanja?
2. When there are no spaces (U+0020) between Hangul and non-ideographic letters/numerals, should the browser insert some extra spacing (like Chinese and Japanese)?

Personally, I think there should be no extra spacing between Hangul and Hanja, but I'm not sure what the answer is to the second question. Comments?

**Comment by anawhj** [comment]
> I think there should be no extra spacing between Hangul and Hanja

Right! There's no extra spacing between Hangul and Hanja, though some cases might show a bit spacing for better styling caused by different font library or system.

When I check the behavior in Han 2025 (No.1 Korean document editor like MS Word), it shows no extra spacing between them.

**Comment by OverflowCat** [comment]
> 2. When there are no spaces (U+0020) between Hangul and non-ideographic letters/numerals, should the browser insert some extra spacing (like Chinese and Japanese)?

I think the answer should be NO as well. IMO the extra spacing should be determined by the semantics of spacing in that language rather than ideo/non-ideo. In Korean, spaces separate a _word cluster_ that consists of a content word (noun or verb stem) plus particles or inflections. So, putting an extra space between the two may lead to ambiguity, and it is not the current practice of Korean language.

I found a good example posted in that csswg issue: https://github.com/prettier/prettier/issues/5028#issuecomment-417825085 You can see the difference between the first two lines:

<ins>Google을</ins> 홈페이지로 설정하기 -> Set <ins>Google as</ins> your homepage (actually Google + object marker `을`)
<ins>Google 설정</ins> 관리	-> Manage <ins>Google settings</ins>

![](https://user-images.githubusercontent.com/6244462/44941341-452d3f80-add6-11e8-996b-687dcff700fc.png)

**Comment by anawhj** [comment]
> https://github.com/prettier/prettier/issues/5028

It looks a great example on this issue.
